docs/proposal-partially-satisfied-prerequisites.md

```diff
@@ -0,0 +1,347 @@
+# Handling of partially satisfied prerequisites
+
+See https://github.com/cylc/cylc-flow/issues/4256
+
+## New trigger syntax
+
+A new trigger syntax is proposed which (it is hoped) will resolve many issues:
+
+* `a:x ?=> b` is an "optional trigger": `b` is optionally triggered by `a`.
```

Comment:
I think that `a:<state> ?=> b` would be more intuitive to the reader.

Reply:
This is referencing the earlier example which used `a:x => b`.